tests/test_config.py

```diff
@@ -175,6 +175,15 @@ def test_cmd_line():
         app = NoConfigApp()
         assert app.cfg.preload_app
 
+
+def test_cmd_line_invalid_setting(monkeypatch):
+    with AltArgs(["prog_name", "-q", "bar"]):
```

Comment:
It would also be nice to test the error message here (e.g. "gunicorn: error: unrecognized arguments: -q")

We can use pytest's `capsys` fixture to do this: http://doc.pytest.org/en/latest/capture.html#accessing-captured-output-from-a-test-function (the py.test documentation is not clear whether the whole feature is added in 3.0 or just `capsys.disabled()` so feel free to update `pytest==2.8.3` in `requirements_test.txt`)